src/components/InngestFunction.ts

```diff
@@ -343,6 +348,7 @@ export class InngestFunction<
            * Whichever the case, this is bad and we can't continue in this
            * undefined state.
            */
+          await logger.flush();
```

Comment:
It looks like every call to `logger.flush()` happens just before the function ends, either from a `throw` or a `return`.

To keep the code paths low (and to make sure we don't miss any in the future), it might be possible to wrap everything in a `try`/`finally` and run `await logger.flush()` in the `finally` block.

```ts
try {
  // everything
  // starting at userFnPromise
  // ending at the end of the fn
} finally {
  await logger.flush();
}
```

Feels a bit grim indenting so many lines, but I'm happy to do so for now to make sure all the code paths are guaranteed to be captured and it's a good signal we need to refactor `runFn` a touch when we can anyway!

Reply:
sure, I'll wrap the `try` after logger got instantiated then